### PR TITLE
ESM module, dependencies upgrade

### PR DESCRIPTION
**Pull request by wisekaa03**

### CODE REVIEW

**Review by mifi — CHANGES_REQUESTED** [review]
Thanks for this PR! Looks good to me, except for a few small things.

Have you tested and it still works? Also is typescript still correct?

**Comment by mifi** [comment]
Oh and we need to rewrite examples to ESM, as well as docs

File: .eslintrc [hunk 1]
```diff
@@ -4,12 +4,14 @@
     "node": true
   },
   "parserOptions": {
-    "sourceType": "script"
+    "sourceType": "module"
   },
+  "import/extensions": [],
```
[the remark this answers is not in the document]
Comment (wisekaa03) [reply .eslintrc:9]: It's for "Unexpected use of file extension "js" for ..."
Maybe you want a
```
 "import/extensions": [".js"],
```

File: cli.js [hunk 2]
```diff
@@ -72,7 +73,7 @@ const cli = meow(`
   };
 
   if (json) {
-    params = JSON5.parse(fs.readFileSync(json, 'utf-8'));
+    params = json5.parse(readFileSync(json, 'utf-8'));
```
[the remark this answers is not in the document]
Comment (wisekaa03) [reply cli.js:76]: Ok

File: package.json [hunk 3]
```diff
@@ -1,35 +1,36 @@
 {
   "name": "editly",
   "description": "Simple, sexy, declarative video editing",
-  "version": "0.12.0",
+  "version": "0.13.0",
```
[the remark this answers is not in the document]
Comment (wisekaa03) [reply package.json:4]: I got it

File: index.js [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
+import { createFrameSource } from './sources/frameSource.js';
+import { parseConfig } from './parseConfig.js';
+import GlTransitions from './glTransitions.js';
+import Audio from './audio.js';
 
 const channels = 4;
 
-
-const Editly = async (config = {}) => {
+export const Editly = async (config = {}) => {
```
[the remark this answers is not in the document]
Comment (wisekaa03) [reply index.js:18]: Ok

File: audio.js [hunk 5]
```diff
@@ -99,7 +99,7 @@ module.exports = ({ ffmpegPath, ffprobePath, enableFfmpegLog, verbose, tmpDir })
         const weights = processedAudioLayers.map(({ audioLayer }) => (audioLayer.mixVolume != null ? audioLayer.mixVolume : 1));
         const args = [
           ...getFfmpegCommonArgs({ enableFfmpegLog }),
-          ...flatMap(processedAudioLayers, ({ layerAudioPath }) => ['-i', layerAudioPath]),
+          ...lodash.flatMap(processedAudioLayers, ({ layerAudioPath }) => ['-i', layerAudioPath]),
```
[the remark this answers is not in the document]
Comment (wisekaa03) [reply audio.js:102]: Ok, but this was a lodash-es

**Comment by wisekaa03** [comment]
All is ok, with typescript.
I use wisekaa03/Editly with my commercial project.

**Comment by mifi** [comment]
thanks! will publish a new version.

**Pull request merged**